jupyterlab/jupyter-ai
# Add base API URL field for Ollama and OpenAI embedding models

## Description

- Fixes #493 
- Fixes #587 
- Fixes #977 
- Fixes #1004 

Jupyter AI currently allows the user to call a model at a URL (location) different from the default one by specifying a selected Base API URL. This can be done for Ollama, OpenAI provider models. However, for these providers, there is no way to change the API URL for embedding models when using the `/learn` command in RAG mode. This PR adds an extra field to make this feasible.

## Testing instructions

Testing as follows for Ollama:
[1] Start the Ollama system from port 11435 instead 11434 (the default): 
`OLLAMA_HOST=127.0.0.1:11435 ollama serve`
[2] Set the Base API URL:
![set base api url](https://github.com/user-attachments/assets/3445f382-3e3f-4542-97b8-ae65f2f7fe6d)
[3] Check that the new API URL works for completions and `/learn`:
